docs/notes/migration_guide.rst

```diff
@@ -0,0 +1,491 @@
+.. include:: ../03-exports/aliases.include
+.. include:: ../03-exports/aliases-api.include
+.. include:: ../03-exports/roles.include
+
+Migration Guide from Fast DDS v2 to Fast DDS v3
+===============================================
+
+This document aims to help during the migration process from eProsima *Fast DDS version* 2 to *Fast DDS version* 3.
+For more information about all the updates, please refer to the [release notes](https://fast-dds.docs.eprosima.com/en/latest/notes/notes.html).
```

Comment:
I would add a reference in supported_versions.rst as `versions_v3` to tag it here
```suggestion
For more information about all the updates, please refer to the :ref:`release notes <release_notes_v3>`.
```